docs/content/en/highlights/2021-07-13-helm-customConfig.md

```diff
@@ -0,0 +1,324 @@
+files added to the "/etc/vector" directory.
+
+### globalOptions and logSchema
+
+Any [global options](https://vector.dev/docs/reference/configuration/global-options/) can be moved directly
+into the `customConfig` key and converted into snake-case.
+
+```diff title="values.yaml"
+   globalOptions:
++    enabled: false
+     dataDir: "/vector-data-dir"
+   logSchema:
++    enabled: false
+     hostKey: "host"
+     messageKey: "message"
+     sourceTypeKey: "source_type"
+     timestampKey: "timestamp"
+   ...
++  customConfig:
++    data_dir: "/vector-data-dir"
++    log_schema:
++      host_key: host
++      message_key: message
++      source_type_key: source_type
++      timestamp_key: timestamp
++  ...
+```
+
+### vectorApi
+
+[Vector API](https://vector.dev/docs/reference/api/) options can be moved as is under a `customConfig.api` key.
+The `extraContainerPorts` or `service` key should be used to expose the port configured in `customConfig`.
+
+```diff title="values.yaml"
+   vectorApi:
++    enabled: false
+     address: "0.0.0.0:8686"
+     playground: true
+   ...
++  customConfig:
++    ...
++    api:
++      enabled: true
++      address: 0.0.0.0:8686
++      playground: true
++    ...
+```
+
+### kubernetesLogsSource
+
+The vector-agent chart will continue to mount the hostPaths required to access Pod logs.
+
+```diff title="values.yaml"
+   kubernetesLogsSource:
++    enabled: false
+     sourceId: kubernetes_logs
+     config: {}
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sources:
++      ...
++      kubernetes_logs:
++        type: kubernetes_logs
++    ...
+```
+
+### vectorSource
+
+The `extraContainerPorts` or `service` key should be used to expose the port configured in `customConfig`.
+
+```diff title="values.yaml"
+   vectorSource:
++    enabled: false
+     sourceId: vector
+     listenAddress: "0.0.0.0"
+     listenPort: "9000"
+     config: {}
+     nodePort: null
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sources:
++      ...
++      vector:
++        type: vector
++        address: 0.0.0.0:9000
++    ...
+```
+
+### vectorSink
+
+```diff title="values.yaml"
+   vectorSink:
++    enabled: false
+     sinkId: vector_sink
+     inputs: ["kubernetes_logs"]
+     host: vector
+     port: "9000"
+     config: {}
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sinks:
++      ...
++      vector_sink:
++        type: vector
++        inputs: ["kubernetes_logs"]
++        address: vector:9000
++    ...
+```
+
+### internalMetricsSource
+
+```diff title="values.yaml"
+   internalMetricsSource:
++    enabled: false
+     sourceId: internal_metrics
+     config: {}
+     rawConfig: null
+   ...
++  customConfig:
++    ...
++    sources:
++      ...
++      internal_metrics:
++        type: internal_metrics
++    ...
+```
+
+### hostMetricsSource
+
+Users should set the PROCFS_ROOT and SYSFS_ROOT environment variables, as well as mounting the
+required hostPaths.
```

Comment:
I could be wrong, but this seems like a lot of to ask of the user to setup this source, right? Is this common practice for other agents that collect host metrics? Is there anything we can do to make this easier for the common case?

Reply:
We could look to recursively check the provided config for `host_metrics` sources and include them if found, or just always set them (metricbeat appears to just always have it)

Reply:
Given we provide them with the commented out config:

https://github.com/timberio/vector/pull/8168/files#diff-45dcaa563e4bba8730fda2e2f7b915381a4e6d803173d27926502df4d25fe5deR154-R174

I think this is fine. I'm not seeing another easy way to improve this that is still consistent with our new approach here of using direct Vector config YAML over custom keys that translate to Vector config.

In the future, we could think about enabling the `host_metrics` source by default for Vector -- I think this would be something like what you are describing? -- but that would be a bit of a shift from the current state where all collection is opted into.

Reply:
Ah, the issue @jszwedko is that they'd also need to include the env vars and mounts to properly collect host metrics

Reply:
Ah, I see. I could see doing that by default, regardless of whether `host_metrics` is used, as long as users can disable it easily if they want.

Reply:
My first preference would be to only do that _if_ they are including a `host_metrics` source, but just always doing it by default would be my second choice.